packages/runtime/lib/runtime.js

```diff
@@ -941,6 +981,14 @@ class Runtime extends EventEmitter {
       if (!silent) {
         this.logger?.info(`Started the ${label}...`)
       }
+
+      const { enabled, gracePeriod } = worker[kConfig].health
+      if (enabled && config.restartOnError > 0) {
+        // unref-ing as we don't want to keep the process alive just because of it
+        setTimeout(() => {
+          this.#setupHealthCheck(worker, label)
+        }, gracePeriod > 0 ? gracePeriod : 1).unref()
```

Comment:
We should cancel this timer after we have got one passing.

Reply:
Can you clarify?

Reply:
Right now there is a race condition, as the runtime can be closed/stopped before those have had the chance to run. Instead of unref(), we should store those somewhere and clear them if they still need to be fired.

I generically think that keeping timers around without cancelling them is problematic. If the runtime is closed before we had a chance of setting up the healtchecks, the timers should be cancelled.

Reply:
I see. Make sense. Will work on it.